frame/fee-market/src/s2s/payment.rs

```diff
@@ -212,8 +216,23 @@ where
 	}
 }
 
+pub(crate) fn cal_slash_value<T: Config>(
+	delay: Option<T::BlockNumber>,
+	locked_collateral: RingBalance<T>,
+) -> RingBalance<T> {
+	let mut slash_value = RingBalance::<T>::zero();
+	if let Some(delay_blocks) = delay {
+		slash_value = UniqueSaturatedInto::<u128>::unique_saturated_into(delay_blocks)
+			.saturating_mul(UniqueSaturatedInto::<u128>::unique_saturated_into(
+				T::SlashPerBlock::get(),
+			))
+			.unique_saturated_into();
+	}
+	sp_std::cmp::min(locked_collateral, slash_value)
+}
+
```

Comment:
This should be a trait that can be configured in runtime, we might want to testing new `S(t)` in tests or testnet.

Reply:
I see, just like the old slasher associated type. Let me give it a change.

Reply:
Done